nl_server/config.py

```diff
@@ -107,80 +98,31 @@ class EmbeddingsConfig:
-      return json.load(f) or {}
-  # If that path doesn't exist, assume we are running locally and use the values
-  # from autopush.
-  else:
-    current_file_path = Path(__file__)
-    autopush_env_values = f'{current_file_path.parent.parent}/deploy/helm_charts/envs/autopush.yaml'
-    with open(autopush_env_values) as f:
-      autopush_env = yaml.full_load(f)
-      return autopush_env['nl']['vertex_ai_models']
-
-
-#
-# Parse the input `embeddings.yaml` dict representation into EmbeddingsInfo
+# Parse the input `embeddings.yaml` dict representation into EmbeddingsConfig
 # object.
 #
-def parse(embeddings_map: Dict[str, any]) -> EmbeddingsConfig:
-  get_vertex_ai_model_info = _get_vertex_ai_model_info()
+def parse(embeddings_map: Dict[str, any], vertex_ai_model_info: Dict[str, any],
+          ranking_enabled: bool) -> EmbeddingsConfig:
   if embeddings_map['version'] == 1:
-    return parse_v1(embeddings_map, get_vertex_ai_model_info)
+    return parse_v1(embeddings_map, vertex_ai_model_info, ranking_enabled)
   else:
     raise AssertionError('Could not parse embeddings map: unsupported version.')
 
 
 #
 # Parses the v1 version of the `embeddings.yaml` dict representation into
-# EmbeddingsInfo object.
+# EmbeddingsConfig object.
 #
-def parse_v1(embeddings_map: Dict[str, any],
-             vertex_ai_model_info: Dict[str, any]) -> EmbeddingsConfig:
-  # parse the models
-  models = {}
-  for model_name, model_info in embeddings_map.get('models', {}).items():
-    model_type = model_info['type']
-    score_threshold = model_info.get('score_threshold',
-                                     constants.SV_SCORE_DEFAULT_THRESHOLD)
-    if model_type == ModelType.LOCAL:
-      models[model_name] = LocalModelConfig(type=model_type,
-                                            score_threshold=score_threshold,
-                                            usage=model_info['usage'],
-                                            gcs_folder=model_info['gcs_folder'])
-    elif model_type == ModelType.VERTEXAI:
-      if model_name not in vertex_ai_model_info:
-        logging.error(
-            f'Could not find vertex ai model information for {model_name}')
-        continue
-      models[model_name] = VertexAIModelConfig(
-          type=model_type,
-          score_threshold=score_threshold,
-          usage=model_info['usage'],
-          project_id=vertex_ai_model_info[model_name]['project_id'],
-          prediction_endpoint_id=vertex_ai_model_info[model_name]
-          ['prediction_endpoint_id'],
-          location=vertex_ai_model_info[model_name]['location'])
-    else:
-      raise AssertionError(
-          'Error parsing information for model {model_name}: unsupported type {model_type}'
-      )
+def parse_v1(embeddings_map: Dict[str, any], vertex_ai_model_info: Dict[str,
+                                                                        any],
+             ranking_enabled: bool) -> EmbeddingsConfig:
```

Comment:
nit: why not pass EmbeddingsSpec

Reply:
EmbeddingsSpec has things not used so felt more clear to pass the things we actully need, wdyt

Reply:
sg for now.